tock/hours/views.py

```diff
@@ -617,43 +616,40 @@ def get_object(self, queryset=None):
         return obj
 
     def clean_task_data(self, float_people_id):
-        # Get Float task data for week corresponding w/ ReportingPeriod
-        # instance.
+        # First day of reporting period.
+        start_date = datetime.datetime.strptime(
+            self.kwargs['reporting_period'], "%Y-%m-%d"
+        ).date()
+        # Last day of reporting period.
+        end_date = start_date + datetime.timedelta(days=6)
+        # Get Float task data for week corresponding start and end dates.
+        # Assumes a reporting period is 1 week.
         r = get_float_data(
             endpoint='tasks',
             params={
-                'weeks': base.FLOAT_API_TASK_WEEKS,
-                'start_day': datetime.datetime.strptime(
-                    self.kwargs['reporting_period'], "%Y-%m-%d"
-                ).date()
+                'weeks': 1,
+                'start_day': start_date
             }
         )
-        if r.status_code != 200:
+        if not r:
             return None
-        else:
-            task_data = r.json()
-        #From response, derive meta data and tasks for the period.
-        f_start_date = datetime.date(
-            int(task_data['start_yr']), 1, 1) + datetime.timedelta(
-            days=int(task_data['start_doy']
-            )-1)
-        f_end_date = f_start_date + datetime.timedelta(
-            days=(base.FLOAT_API_TASK_WEEKS * 6))
+        task_data = r.json()['people']
         clean_data = {
-            'metadata': {
-                'float_period_start': f_start_date,
-                'float_period_end': f_end_date
-            },
-            'tasks': [ item for sublist in \
-                [ ii for ii in [i['tasks'] for i in task_data['people'] \
-                if i['people_id'] == float_people_id] ] \
-            for item in sublist ]
+            # Get all of the tasks (ii) associated with a Float user if the
+            # user's Float people_id matches their Tock float_people_id.
+            # See https://github.com/floatschedule/api/blob/805663be98c9f48a275c9a13e824b0d6701df398/Sections/tasks.md.
+            'tasks': flatten(
+                [ ii for ii in [i['tasks'] for i in task_data \
+                    if int(i['people_id']) == float_people_id] ]
```

Comment:
Er wait... didn't you say that sometimes the Tock API has an empty string for `people_id` or something?  Will that cause the `int()` call to throw a `ValueError`?